sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/tests/TimeSeriesIdTests.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using FluentAssertions;
+using NUnit.Framework;
+
+namespace Azure.Iot.TimeSeriesInsights.Tests
+{
+    public class TimeSeriesIdTests : E2eTestBase
+    {
+        // TODO: replace hardcoding the Type GUID when the Types resource has been implemented
+        private const string DefaultType = "1be09af9-f089-4d6b-9f0b-48018b5f7393";
+        private const int MaxNumberOfRetries = 10;
+
+        private readonly TimeSpan _delay = TimeSpan.FromSeconds(10);
+
+        public TimeSeriesIdTests(bool isAsync)
+            : base(isAsync)
+        {
+        }
+
+        [Test]
+        public async Task TimeSeriesInsightsInstances_TimeSeries3Id()
+        {
+            // Arrange
+            TimeSeriesInsightsClient client = GetClient();
+
+            // Create a Time Series Id with 3 properties. Middle property is a null
+            ITimeSeriesId idWithNull = new TimeSeries3Id(
+                Recording.GenerateAlphaNumericId(string.Empty, 5),
+                null,
+                Recording.GenerateAlphaNumericId(string.Empty, 5));
+
+            var timeSeriesInstances = new List<TimeSeriesInstance>
+            {
+                new TimeSeriesInstance(idWithNull, DefaultType),
+            };
+
+            // Act and assert
+            try
+            {
+                // Create TSI instances
+                Response<InstancesOperationError[]> createInstancesResult = await client
+                    .CreateOrReplaceTimeSeriesInstancesAsync(timeSeriesInstances)
+                    .ConfigureAwait(false);
+
+                // Assert that the result error array does not contain any object that is set
+                createInstancesResult.Value.All((errorResult) => errorResult == null)
```

Comment:
After a couple of hours of thinking about it, I finally got my head wrapped around the previous approach (`Any`, `!= null`, `BeFalse`). I don't know why that is so hard for me.

Reply:
Since we discussed this  yesterday, I like it better this way. Now I can't wrap my head around the way I had it before :)